src/averageAges.js

```diff
@@ -37,7 +42,16 @@ function calculateMenAverageAge(people, century) {
  * @return {number}
  */
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  const women = withChildren
+    ? people.filter((p) => p.sex === 'f'
+      && people.find((p1) => p1.mother === p.name))
```

Comment:
Using find inside filter leads to a nested loop, which can be inefficient for large datasets. 

Create a Set of all mothers, then filter. 

This reduces the time complexity from O(n^2) to O(n).